SAT/Z3Interfacing.cpp

```diff
@@ -76,18 +76,34 @@ struct BottomUpChildIter<z3::expr>
 
 
 auto quotient0_name(char c, z3::sort s) 
-{ return outputToString("quot-0-", c, "-", s); }
+{ return Output::toString("quot-0-", c, "-", s); }
 
 auto remainder0_name(char c, z3::sort s) 
-{ return outputToString("rem-0-", c, "-", s); }
+{ return Output::toString("rem-0-", c, "-", s); }
 
 template<class UInt64ToExpr>
 z3::expr int_to_z3_expr(IntegerConstantType const& val, UInt64ToExpr toExpr) {
     auto sign = val.sign();
-    auto abs = val.abs().toInner();
+    auto abs = val.abs();
+
+    // TODO recycled stack
+    Stack<uint64_t> digits;
+    z3::expr base =  // <- == 2^64
+      toExpr(std::numeric_limits<uint64_t>::max()) + toExpr(1);
+    while(!abs.fits<unsigned long>()) {
```

Comment:
Is this how we communicate out arbitrary large numbers to z3's arbitrary large numbers?

Reply:
Yes exactly!

Reply:
Added a comment just now.